src/rules/no-reaching-inside.js

```diff
@@ -0,0 +1,100 @@
+import find from 'lodash.find'
+import minimatch from 'minimatch'
+
+import resolve from '../core/resolve'
+import importType from '../core/importType'
+import isStaticRequire from '../core/staticRequire'
+
+module.exports = function noReachingInside(context) {
+  const options = context.options[0] || {}
+  const allowRegexps = (options.allow || []).map(p => minimatch.makeRe(p))
+
+  // test if reaching to this destination is allowed
+  function reachingAllowed(importPath) {
+    return !!find(allowRegexps, re => re.test(importPath))
+  }
+
+  // minimatch patterns are expected to use / path seperators, like import
```

Comment:
I spell that word wrong everywhere, all the time, I think I need to setup autohotkey just for it.